lib/tasks/stats.rake

```diff
@@ -19,4 +19,13 @@ namespace :stats do
       DatadogApi.gauge('efile_submissions.state_counts', count, tags: ["current_state:#{state}"])
     end
   end
+
+  desc "Monitor the longest periods of time that any efile submission has been in preparing, bundling, and queued"
+  task monitor_delayed_efile_submissions: :environment do
+    [:preparing, :bundling, :queued].each do |state|
+      oldest_transition_to = EfileSubmissionTransition.where(most_recent: true, to_state: state).sort_by(&:created_at).first
```

Comment:
gah, sorry to take another shot at this _but_ i wonder if there is a performance differnce by doing an enumerable sort_by and getting the first item vs doing a .order(:created_at).limit(1), which would only retrieve the first item by created_at out of the db instead of all of them and then sorting them.

I think we're likely fine to merge, but it might be an interesting query to try to benchmark and see which is more performant.

Reply:
mm, not sure! i honestly don't know how to test that but would be happy to pair on it if you want? i might merge this first anyway but we can refactor in a follow-up commit